src/TeamStore.DevUI/Factories/AssetFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace TeamStore.Factories
+{
+    using System.Collections.Generic;
+    using TeamStore.DevUI.ViewModels;
+    using TeamStore.Keeper.Models;
+
+    // These method names need to be renamed to follow a consistent pattern
+    public static class AssetFactory
+    {
+        public static List<AssetSearchViewModel> ConvertAssetSearch(List<Asset> assets)
+        {
+            var result = new List<AssetSearchViewModel>(assets.Count);
+            foreach (var asset in assets)
+            {
+                result.Add(CreateAssetSearchViewModel(asset));
+            }
+
+            return result;
+        }
+
+        public static AssetSearchViewModel CreateAssetSearchViewModel(Asset result)
+        {
+            var viewModel = new AssetSearchViewModel();
```

Comment:
*every* method in every programming language *should* start by validating input and with unit tests to prove that you are validating input.

Reply:
I can't agree more, nevertheless can't think of applicable validation rules. The only limitation would be for the primary and foreign key to be non-negative. They are extracted from the database though. C# strong typing should be enough for this case. Please let me know if I misinterpret the issue.